File: java/CarreraMencion.java
package java;
import java.util.Objects;

public class CarreraMencion extends Carrera{
    private PilaMencion a;


    public CarreraMencion() {
        super();
    }

    public CarreraMencion(String nombre, int planEstudios, int idCarrera, PilaMencion a) {
        super(nombre, planEstudios, idCarrera);
        this.a = a;
    }

    public PilaMencion getA() {
        return this.a;
    }

    public void setA(PilaMencion a) {
        this.a = a;
    }

    public CarreraMencion a(PilaMencion a) {
        setA(a);
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof CarreraMencion)) {
            return false;
        }
        CarreraMencion carreraMencion = (CarreraMencion) o;
        return Objects.equals(a, carreraMencion.a);
    }



    @Override
    public String toString() {
        return super.toString()+"{" +
            " a='" + getA() + "'" +
            "}";
    }
    
}

File: c#/CarreraMencion.cs
using System;
using System.Collections;
using System.Linq;
using System.Threading.Tasks;


using System.IO;
namespace c_;
[Serializable]

public class CarreraMencion: Carrera
{
    private Stack<string> menciones;

    public CarreraMencion() {
        menciones=new Stack<string>();
    }

    public CarreraMencion(string nombre, int planEstudios, int idCarrera, Stack<string> menciones): base(nombre, planEstudios, idCarrera) {
        this.menciones = menciones;
    }
    
    public Stack<string> getA() {return this.menciones;}
    public void setA(Stack<string> menciones) {this.menciones = menciones;}

    public CarreraMencion a(Stack<string> a) {
        setA(a);
        return this;
    }
    public new void leer(){
        base.leer();
        Console.WriteLine("- Lectura Menciones");
		do {
            Console.Write("\t>> Inserte mencion => ");
            string men=Leer.Cadena;
            menciones.Append(men);
			Console.WriteLine(">>>>>> Desea seguir añadiendo menciones? (s/n) => ");
    	}while(Console.ReadKey().KeyChar=='s');
    } 

    public void mostrarMenciones(){
        IEnumerator<string> r=menciones.GetEnumerator();
        while(r.MoveNext()){
            Console.WriteLine("\t-"+r.Current);
        }
    } 

    //@Override
    /*
    public bool equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof CarreraMencion)) {
            return false;
        }
        CarreraMencion carreraMencion = (CarreraMencion) o;
        return Objects.equals(menciones, carreraMencion.menciones);
    }
    */

        
}
